### PR TITLE
feat: Edit and delete project

### PR DESCRIPTION
**Pull request by revengemiroz**
# Description

Create edit project feature, delete project and cache updates on both of them

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

## Showcase

Please attach a picture or a video showing what this change does.

### CODE REVIEW

**Comment by ivanms1** [comment]
Task linked: [CU-3e5rhfk Edit Project Page](https://app.clickup.com/t/3e5rhfk)

File: apps/web/pages/project-edit/[id].tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import ProjectEdit from '@/pages/ProjectEdit';
```
Comment (ivanms1) [inline apps/web/pages/project-edit/[id].tsx:1]: no ssr needed on this page

File: apps/web/src/components/Dropzone/index.tsx [hunk 2]
```diff
@@ -31,11 +31,17 @@ function Dropzone({
       <input {...getInputProps()} />
       {currentFile ? (
         withPreview ? (
-          <Image
-            src={URL.createObjectURL(currentFile)}
-            alt={currentFile?.name}
-            layout='fill'
-          />
+          <>
```
Comment (ivanms1) [inline apps/web/src/components/Dropzone/index.tsx]: remove this fragment

File: apps/web/src/components/Dropzone/index.tsx [hunk 3]
```diff
@@ -31,11 +31,17 @@ function Dropzone({
...
+            <Image
+              src={
+                typeof currentFile == 'string'
```
Comment (ivanms1) [inline apps/web/src/components/Dropzone/index.tsx]:
```suggestion
                typeof currentFile === 'string'
```

File: apps/web/src/pages/CreateProject/CreateProject.tsx [hunk 4]
```diff
@@ -1,16 +1,19 @@
-import React, { useState } from 'react';
```
Comment (ivanms1) [inline apps/web/src/pages/CreateProject/CreateProject.tsx:1]: Create a separate component named `ProjectForm` that has the following props, `onSubmit` `initialValues` and anything you need. Instead of combining all the logic here. This is the `CreateProject` page, should only have logic for that page.

File: apps/web/src/pages/CreateProject/DetailsFormModal/index.tsx [hunk 5]
```diff
@@ -87,7 +91,7 @@ const DetailsFormModal = ({
           disabled={isLoading}
           onClick={handleSubmit(onSubmit)}
         >
-          Publish
+          {router.pathname.includes('/project-edit/') ? 'Update' : 'Publish'}
```
Comment (ivanms1) [inline apps/web/src/pages/CreateProject/DetailsFormModal/index.tsx]: change to `Submit` same for both cases

File: apps/web/src/pages/Project/helper.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+export const checkValidationForProjectOptions = (currentUser, projectUser) => {
```
Comment (ivanms1) [inline apps/web/src/pages/Project/helper.js]: change it to a ts and use a hook instead, `useIsProjectAuthor.ts`
```ts
import useIsLoggedIn from '@/hooks/useIsLoggedIn';

const useIsProjectAuthor = (authorId) => {
  const { currentUser } = useIsLoggedIn();

  return currentUser?.id === authorId;
};

export default useIsProjectAuthor;
```

Comment (ivanms1) [reply apps/web/src/pages/Project/helper.js]: create inside the hooks folder

File: apps/web/src/pages/Project/index.tsx [hunk 7]
```diff
@@ -23,30 +24,81 @@ import {
...
+
+import { checkValidationForProjectOptions } from './helper';
 
 function Project() {
   const router = useRouter();
   const { query } = useRouter();
 
-  const { data } = useGetProjectQuery({
+  const notifySuccess = () => toast.success('Project deleted successfully');
+  const notifyFailure = () => toast.error('Project deletetion failed');
```
Comment (ivanms1) [inline apps/web/src/pages/Project/index.tsx:41]: if these are only use in one place, just use the toast directly instead on creating variables.

File: apps/web/src/pages/Project/index.tsx [hunk 8]
```diff
@@ -104,6 +156,23 @@ function Project() {
...
+          <ProjectOptions>
+            <Link href={`/project-edit/${query?.id}`}>
+              <a>
+                <Button variant='ghost'>Edit</Button>
+              </a>
+            </Link>
+
+            <Button
+              variant='ghost'
+              onClick={() => deleteProjectClick(data?.project?.id)}
```
Comment (ivanms1) [inline apps/web/src/pages/Project/index.tsx]: please add a confirmation modal if possible.

File: apps/web/src/pages/ProjectEdit/ProjectEdit.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  const { data } = useGetProjectQuery({
+    variables: {
+      id: String(query.id),
+    },
+    skip: !query.id,
+  });
+
+  return (
+    <>
+      <CreateProject projectDetails={data?.project} />
```
Comment (ivanms1) [inline apps/web/src/pages/ProjectEdit/ProjectEdit.tsx]: Create the `ProjectForm` component I mentioned before and use it here.

File: apps/web/src/pages/User/index.tsx [hunk 10]
```diff
@@ -104,13 +104,13 @@ const User = () => {
...
         )}
         <StyledTitle>{user?.name}</StyledTitle>
+        <FollowButton onClick={handleFollowUser}>
+          {isFollowingData?.user?.isFollowing ? 'Unfollow' : 'Follow'}
+        </FollowButton>
+        <h4>{isFollowingData?.user?.followerCount} Followers</h4>
       </StyledUserContainer>
       {user ? (
         <StyledProjectContainer>
-          <FollowButton onClick={handleFollowUser}>
```
Comment (ivanms1) [inline apps/web/src/pages/User/index.tsx:110]: why was this moved?

File: packages/ui/Modal/index.tsx [hunk 11]
```diff
@@ -7,16 +7,19 @@ import { modal, overlay, StyledModal } from './styles';
...
 export const Modal = ({
   children,
   onClose,
   className,
+  modalzIndex,
   ...props
 }: ModalProps) => {
   return (
     <StyledModal
+      modalzIndex={modalzIndex}
```
Comment (ivanms1) [inline packages/ui/Modal/index.tsx]: remove, pass it in the className

**Review by ivanms1 — CHANGES_REQUESTED** [review]

File: apps/web/src/components/Dropzone/index.tsx (hunk 2 above)
Comment (revengemiroz) [reply apps/web/src/components/Dropzone/index.tsx]: done

File: apps/web/src/components/Dropzone/index.tsx (hunk 3 above)
Comment (revengemiroz) [reply apps/web/src/components/Dropzone/index.tsx]: done

File: apps/web/pages/project-edit/[id].tsx (hunk 1 above)
Comment (revengemiroz) [reply apps/web/pages/project-edit/[id].tsx:1]: done

File: apps/web/src/pages/CreateProject/DetailsFormModal/index.tsx (hunk 5 above)
Comment (revengemiroz) [reply apps/web/src/pages/CreateProject/DetailsFormModal/index.tsx]: done

File: apps/web/src/pages/User/index.tsx (hunk 10 above)
Comment (revengemiroz) [reply apps/web/src/pages/User/index.tsx:110]: seems like a bad placement for the button UI wise, which was on top of the project Grid seems resonable to be grouped together with the user avatar pic, name an followers count

File: apps/web/src/pages/Project/index.tsx (hunk 8 above)
Comment (revengemiroz) [reply apps/web/src/pages/Project/index.tsx]: done

**Review by ivanms1 — APPROVED** [review]
LGTM
